python/pyspinw/plotting.py

```diff
@@ -0,0 +1,475 @@
+            label = swobj.unit_cell['label'][atom_idx-1]
+            size = matlab_caller.sw_atomdata(label, 'radius')[0,0]
+            unit_cell.add_atom(AtomSimple(atom_idx, swobj.atom()['r'][:,iatom], S=S, size=size, color=color, label=label,
+                                          gtensor_mat=g_mat, aniso_mat=aniso_mat))
+            
+        # add bonds - only plot bonds for which there is a mat_idx
+        bond_idx = np.squeeze(swobj.coupling['idx'])
+        bond_matrices = swobj.matrix['mat'].reshape(3,3,-1)
+        for ibond in np.unique(bond_idx[np.any(swobj.coupling['mat_idx'], axis=0)]):
+            i_dl = np.squeeze(bond_idx==ibond)
+            for mat_idx in swobj.coupling['mat_idx'][:, np.argmax(i_dl)]:
+                if mat_idx > 0:
+                    unit_cell.add_bond_vertices(f"bond{ibond}_mat{mat_idx}", np.squeeze(swobj.coupling['atom1'])[i_dl]-1, np.squeeze(swobj.coupling['atom2'])[i_dl]-1, swobj.coupling['dl'].T[i_dl],
+                                                bond_matrices[:,:,mat_idx-1], color=swobj.matrix['color'][:,mat_idx-1]/255)
+        
+        # generate unit cells in supercell
+        self.unit_cells = []
+        self.extent = np.asarray(extent)
+        self.int_extent = np.ceil(self.extent).astype(int) + 1  # to plot additional unit cell along each dimension to get atoms on cell boundary
+        for zcen in range(self.int_extent[2]):
+            for ycen in range(self.int_extent[1]):
+                for xcen in range(self.int_extent[0]):
+                    self.unit_cells.append(unit_cell.translate_by([xcen, ycen, zcen]))
+                    
+        # get magnetic structure for all spins in supercell
+        self.n = np.zeros(3)
+        self.apply_magnetic_structure(swobj)
+        
+        # transforms
+        self.basis_vec = unit_cell.basis_vec
+        self.inv_basis_vec = np.linalg.inv(self.basis_vec)
+                
+        # scale factors
+        self.abc = np.sqrt(np.sum(self.basis_vec**2, axis=1))
+        self.cell_scale_abc_to_xyz = min(self.abc)
+        self.supercell_scale_abc_to_xyz = min(self.abc*self.extent)
+        self.bond_width = 5
+        self.spin_scale = 1
+        self.arrow_width = 8
+        self.arrow_head_size = 5
+        self.font_size = 20
+        self.axes_font_size = 50
+        self.atom_alpha = 0.5
+        self.rotation_plane_radius = 0.3*self.cell_scale_abc_to_xyz
+        self.ion_radius = 0.3*self.cell_scale_abc_to_xyz
+        self.dm_arrow_scale = 0.2*self.cell_scale_abc_to_xyz
+
+    def transform_points_abc_to_xyz(self, points):
+        return points @ self.basis_vec
+        
+    def transform_points_xyz_to_abc(self, points):
+        return points @ self.inv_basis_vec
+
+    def apply_magnetic_structure(self, swobj):
+        magstr = swobj.magstr('NExt', [int(ext) for ext in self.int_extent])
+        mj = magstr['S']
+        if not np.any(mj):
+            print('No magnetic structure defined')
```

Comment:
It might be better to turn these `print` calls (this, the polyhedron warning on line 363 and the singular anisotropy matrix in line 456) into `warnings.RuntimeWarning` or similar which can be suppressed...